src/components/include/utils/threads/message_loop_thread.h

```diff
@@ -148,8 +148,9 @@ MessageLoopThread<Q>::MessageLoopThread(const std::string& name,
 template <class Q>
 MessageLoopThread<Q>::~MessageLoopThread() {
   Shutdown();
-  delete thread_delegate_;
   threads::DeleteThread(thread_);
+  thread_ = NULL;
```

Comment:
@LuxoftAKutsan Why do you set thread_ pointer to `NULL` after deletion but did not set `thread_delegate_`? It is a pointer as well.